shotover-proxy/src/message/mod.rs

```diff
@@ -345,26 +350,33 @@ impl Value {
             match spec.col_type.id {
                 ColType::Ascii => Value::Strings(decode_ascii(actual_bytes).unwrap()),
                 ColType::Bigint => Value::Integer(decode_bigint(actual_bytes).unwrap()),
-                ColType::Blob => Value::Bytes(Bytes::copy_from_slice(actual_bytes)),
                 ColType::Boolean => Value::Boolean(decode_boolean(actual_bytes).unwrap()),
-                ColType::Counter => Value::Integer(decode_int(actual_bytes).unwrap() as i64),
-                ColType::Decimal => unimplemented!("We dont have a decimal type yet"),
+                ColType::Counter | ColType::Int => {
+                    Value::Integer(decode_int(actual_bytes).unwrap() as i64)
+                }
+                ColType::Decimal => todo!(),
                 ColType::Double => Value::Float(decode_double(actual_bytes).unwrap()),
                 ColType::Float => Value::Float(decode_float(actual_bytes).unwrap() as f64),
-                ColType::Int => Value::Integer(decode_int(actual_bytes).unwrap() as i64),
-                ColType::Uuid => Value::Bytes(Bytes::copy_from_slice(actual_bytes)),
                 ColType::Varchar => Value::Strings(decode_varchar(actual_bytes).unwrap()),
-                ColType::Varint => unimplemented!("We dont have a varint type yet"),
-                ColType::Timeuuid => Value::Bytes(Bytes::copy_from_slice(actual_bytes)),
+                ColType::Varint => todo!(),
+                ColType::Timeuuid | ColType::Blob | ColType::Uuid => {
+                    Value::Bytes(Bytes::copy_from_slice(actual_bytes))
+                }
                 ColType::Inet => Value::Inet(decode_inet(actual_bytes).unwrap()),
-                ColType::Timestamp => Value::NULL,
-                ColType::Date => Value::NULL,
-                ColType::Time => Value::NULL,
+                ColType::Date | ColType::Timestamp | ColType::Time => Value::NULL, // TODO
```

Comment:
Date and Time are represented as integers in Cassandra, should they be stored in the Int variant or make a separate one

Reply:
If we stored them as ints I dont think we would have a way of telling that they were originally dates.

Reply:
I'm happy for the discriminate on the enum to be the type tag here, but we probably need to make sure we have enough fidelity to represent the most precise version of this... and it's something easily convertible to other time formats